OutOfSchool/OutOfSchool.DataAccess/Repository/WorkshopRepository.cs

```diff
@@ -45,5 +46,23 @@ public new async Task Delete(Workshop entity)
 
         /// <inheritdoc/>
         public bool ClassExists(long id) => db.Classes.Any(x => x.Id == id);
+
+        /// <inheritdoc/>
+        public async Task<IEnumerable<Workshop>> GetListOfWorkshopsForSynchronizationByOperation(ElasticsearchSyncOperation operation)
+        {
+            var elasticsearchSyncRecords = db.ElasticsearchSyncRecords;
+
+            var resultMaxDates = from record in elasticsearchSyncRecords
+                                 group record by record.RecordId into groupedRecords
+                                 select new { RecordId = groupedRecords.Key, MaxOperationDate = groupedRecords.Max(r => r.OperationDate) };
+
+            var result = from rMaxDate in resultMaxDates
+                         join record in elasticsearchSyncRecords on new { rMaxDate.RecordId, OperationDate = rMaxDate.MaxOperationDate } equals new { record.RecordId, record.OperationDate } into leftJoin
+                         from joinedRecord in leftJoin
+                         where joinedRecord.Operation == operation
```

Comment:
at first part of query I take latest records by date and I don't need filter it by operation beforehand because it possible that one I could be created, updated or deleted during this period. then I join with the same table for getting not aggregated information (operation in this case). It was first iteration of implementation of this query. I will think about its optimization.